tests/languages/julia_test.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from pre_commit.languages import julia
+from testing.language_helpers import run_language
+
+
+def _make_hook(tmp_path, julia_code):
+    src_dir = tmp_path.joinpath('src')
+    src_dir.mkdir()
+    src_dir.joinpath('main.jl').write_text(julia_code)
+    tmp_path.joinpath('Project.toml').write_text(
+        '[deps]\n'
+        'Example = "7876af07-990d-54b4-ab0e-23690620f79a"\n',
+    )
+
+
+def test_julia_hook(tmp_path):
+    code = """
+    using Example
+    function main()
+        println("Hello, world!")
+    end
+    main()
+    """
+    _make_hook(tmp_path, code)
+    expected = (0, b'Hello, world!\n')
+    assert run_language(tmp_path, julia, 'src/main.jl') == expected
+
+
+def test_julia_hook_args(tmp_path):
+    code = """
+    function main(argv)
+        foreach(println, argv)
+    end
+    main(ARGS)
+    """
+    _make_hook(tmp_path, code)
+    expected = (0, b'--arg1\n--arg2\n')
+    assert run_language(
+        tmp_path, julia, 'src/main.jl --arg1 --arg2',
+    ) == expected
+
+
+def test_julia_hook_additional_deps(tmp_path):
+    code = """
+    using TOML
+    function main()
+        project_file = Base.active_project()
+        dict = TOML.parsefile(project_file)
+        for (k, v) in dict["deps"]
+            println(k, " = ", v)
+        end
+    end
+    main()
+    """
+    _make_hook(tmp_path, code)
+    deps = ('TOML=fa267f1f-6049-4f14-aa54-33bafae1ed76',)
+    ret, out = run_language(tmp_path, julia, 'src/main.jl', deps=deps)
+    assert ret == 0
+    assert b'Example = 7876af07-990d-54b4-ab0e-23690620f79a' in out
+    assert b'TOML = fa267f1f-6049-4f14-aa54-33bafae1ed76' in out
```

Comment:
maybe I just don't know enough about julia but from a quick search it seems like TOML is a stdlib library? does it still need to be installed?

Reply:
It is a standard library, and it can be loaded at top level (e.g. in the Julia REPL) by default, since the `@stdlib` entry is included in the default `LOAD_PATH`. However, for Julia packages it is required (so people are used to explicitly adding also stdlibs), and for full reproducibility it is recommended. There is also ongoing work to separate stdlibs to make them less tied to the Julia version, so then they would behave more like normal packages that can be upgraded separately etc.

We can of course make the choice to include `@stdlib` in the load path for julia hooks, but I would argue that it is better to be explicit with the dependencies in this case.